xdsl/dialects/dlti.py

```diff
@@ -30,13 +41,31 @@ class DataLayoutEntryAttr(ParametrizedAttribute):
     key: Attribute
     value: Attribute
 
+    def __init__(
+        self,
+        key: StringAttr | TypeAttribute | str,
+        value: Attribute | str | int | DictValueType,
+    ):
+        if isinstance(key, str):
+            key = StringAttr(key)
+
+        if isinstance(value, str):
+            value = StringAttr(value)
+        elif isinstance(value, int):
+            value = IntAttr(value)
+        elif isinstance(value, Mapping):
+            value = MapAttr(value)
+
+        super().__init__(key, value)
```

Comment:
Any thoughts on trying the new `converter` functionality here?

Reply:
> Any thoughts on trying the new `converter` functionality here?

what's that?

Reply:
See the builtin `tuple` type. Essentially each parameter can define a `converter` method which essentially does these `if isinstance(key, str): key = StringAttr(key)` clauses. We haven't really explored this fully yet as we added it quite recently.

I'll leave it up to you whether you want to try it, it's always something we can change later/iterate on if wanted.

Reply:
done :)